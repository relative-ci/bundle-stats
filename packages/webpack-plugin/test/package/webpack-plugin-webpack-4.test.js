const webpack4 = require('webpack4');
const MemoryFS = require('memory-fs');
const { advanceTo } = require('jest-date-mock');

// eslint-disable-next-line global-require
jest.mock('webpack', () => require('../../../../node_modules/webpack4'));

const config = require('./app/webpack.config');

advanceTo(new Date(2020, 10, 30));

jest.setTimeout(10 * 1000);

describe('webpack plugin package', () => {
  test('webpack4', (done) => {
    expect.assertions(3);

    const compiler = webpack4(config);
    compiler.outputFileSystem = new MemoryFS();

    compiler.run((error, stats) => {
      expect(error).toEqual(null);
      expect(stats.hasErrors()).toBe(false);
      expect(stats.toJson({ source: false, assets: true }).assets).toMatchSnapshot();
      done();
    });
  });
});
