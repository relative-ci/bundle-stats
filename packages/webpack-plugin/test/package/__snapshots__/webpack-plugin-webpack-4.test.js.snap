// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack plugin package webpack4 1`] = `
[
  {
    "chunkNames": [],
    "chunks": [],
    "emitted": true,
    "info": {},
    "isOverSizeLimit": undefined,
    "name": "bundle-stats.html",
    "size": 0,
  },
  {
    "chunkNames": [
      "main",
    ],
    "chunks": [
      0,
    ],
    "emitted": true,
    "info": {},
    "isOverSizeLimit": undefined,
    "name": "main.js",
    "size": 956,
  },
]
`;
